graphql/database/base/RedisModel.js

```diff
@@ -0,0 +1,107 @@
+import Redis from 'ioredis'
+import Model from './Model'
+import {
+  DatabaseConditionalCheckFailedException,
+  DatabaseItemDoesNotExistException,
+  FieldDoesNotExistException,
+  UnauthorizedQueryException,
+} from '../../utils/exceptions'
+
+const client = new Redis(process.env.UPSTASH_HOST)
+
+class RedisModel extends Model {
+  static async getInternal(key) {
+    const redisKey = this.getRedisKey(key)
+    const item = await client.hgetall(redisKey)
+    if (Object.keys(item).length === 0) {
+      throw new DatabaseItemDoesNotExistException()
+    }
+    return this.wrapItem(item)
+  }
+
+  static async getBatchInternal(keys) {
+    const items = keys.map(key => this.getInternal(key))
+    return Promise.all(items)
+  }
+
+  static async createInternal(item, overwrite = false) {
+    const redisKey = this.getRedisKey(item.id)
```

Comment:
@jedtan Same question as below: should the "id" property actually be the hash key?

Reply:
Yep, that's correct.